tutorials/haskell/haskell-time-tutorial/tutorial.md

```diff
@@ -0,0 +1,523 @@
+The [MonthOfYear][month-pattern] pattern allows to define months of the year
+not by a number but by its name.
+
+The `MonthOfYear` data type is defined like this:
+
+```haskell
+type MonthOfYear = Int
+```
+
+As you can see, `MonthOfYear` is only an alias for `Int`. Now, it's possible to
+define months like `Int`s, where January will correspond to 1 and so on. But
+fortunately, since GHC 7.8 it's feasible to use pattern synonyms and, instead of
+using raw `Int`s is possible to write this down in a pattern style and have a
+more idiomatic way of defining months:
+
+```haskell
+pattern January :: MonthOfYear
+pattern January = 1
+
+pattern February :: MonthOfYear
+pattern February = 2
+
+pattern March :: MonthOfYear
+pattern March = 3
+.
+.
+.
+
+-- | The twelve 'MonthOfYear' patterns form a @COMPLETE@ set.
+pattern December :: MonthOf
+pattern December = 12
+
+{-# COMPLETE January, February, March, April, May, June, July, August, September, October, November, December #-}Year
+```
+
+If you’re curious about why the `{-# COMPLETE #-}` pragma is being
+used, see the [COMPLETE pragma documentation][complete-pragma] for more
+information.
+
+Just to make sure that `November` and `11` are equivalent, let's test it
+in a repl:
+
+```bash
+> November == 11
+True
+```
+
+To exemplify, this pattern will be used in a function that receives a `Day` and
+yields if the corresponding aate is a holiday:
+
+```haskell
+dayToHoliday :: Day -> String
+dayToHoliday (YearMonthDay _ January  _)  = "Happy new year!"
+dayToHoliday (YearMonthDay _ November 1)  = "Let's eat colada morada"
+dayToHoliday (YearMonthDay _ December 25) = "HO HO HO Merry Christmas"
+dayToHoliday _                            = "Probably just a regular day"
+```
+
+Another pattern synonym is introduced here. The `YearMonthDay` pattern builds
+a `Day` in terms of the `Year`, `MonthOfYear` (do you remember this one?) and
+`DayOfMOnth`, where all three are basically `Int`s. Here is its definition:
+
+```haskell
+pattern YearMonthDay :: Year -> MonthOfYear -> DayOfMonth -> Day
+pattern YearMonthDay y m d <-
+   (toGregorian -> (y, m, d))
+   where
+       YearMonthDay y m d = fromGregorian y m d
+```
+
+Back to the `dayToHoliday` function, let's check that is working:
+
+```haskell
+> dayToHoliday (YearMonthDay 2021 11 01)
+"Let's eat colada morada"
+
+> dayToHoliday (YearMonthDay 2021 12 25)
+"HO HO HO Merry Christmas"
+```
+
+(Now you're probably wondering what “colada morada” is? Haha no worries, it’s an
+[Ecuatorian meal][colada-morada] that is purple corn based with spices and
+fruits, pretty amazing to be honest.)
+
+Back to Haskell, probably it’s pretty clear what the `dateToHoliday` function
+does, the first clause matches all days of January, second one matches the All
+Saints holiday, the third one matches for Christmas and the last one it’s the
+fallback for all other days. By using the `MonthOfYears` pattern, it makes it a
```

Comment:
Is it `YearMonthDay` instead of `MonthOfYears` here?

Reply:
Actually are both, I will rephrase this.